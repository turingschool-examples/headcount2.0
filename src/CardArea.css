.card-area {
  display: flex;
  flex-wrap: wrap;
  margin: 0 auto;
  padding-left: 2rem;
  max-width: 110rem;
  color:rgb(255, 222, 173);
}

@media only screen and (max-width: 1100px) {
  .card-area {
    max-width: 75rem;
  }
}

@media only screen and (max-width: 790px) {
  .card-area {
    max-width: 38rem;
  }
}