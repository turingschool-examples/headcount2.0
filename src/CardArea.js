import React from 'react';
import Card from './Card';
import './CardArea.css';
import PropTypes from 'prop-types';

const CardArea = (props) => {
  let cardObjectList = Array.from(props.data);

  const districtCards = cardObjectList.map((cardObj, index) =>
    <Card key={index} {...cardObj} selectCard={props.selectCard} selectedCards={props.selectedCards}/>
  );

  return (
    <div className="card-area">
      {districtCards}
    </div>
  );
};

CardArea.propTypes = {
  props: PropTypes.objectOf(PropTypes.shape({
    data: PropTypes.object,
    selectCard: PropTypes.func,
    selectedCards: PropTypes.array
  })),
  selectCard: PropTypes.func,
  selectedCards: PropTypes.array
};

export default CardArea;