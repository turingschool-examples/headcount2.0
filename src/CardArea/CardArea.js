import React from 'react';
import Card from '../Card/Card';
import './CardArea.css';
import PropTypes from 'prop-types';

const CardArea = (props) => {
  let cardObjectList = Array.from(props.cleanData);

  const districtCards = cardObjectList.map((cardObj, index) =>
    <Card 
      key={index} 
      {...cardObj} 
      selectCard={props.selectCard} 
      selectedCards={props.selectedCards}
    />
  );

  return (
    <div className="card-area">
      {districtCards}
    </div>
  );
};

CardArea.propTypes = {
  props: PropTypes.objectOf(PropTypes.shape({
    cleanData: PropTypes.object.isRequired,
    selectCard: PropTypes.func.isRequired,
    selectedCards: PropTypes.array.isRequired
  })),
  cleanData: PropTypes.array.isRequired,
  selectCard: PropTypes.func.isRequired,
  selectedCards: PropTypes.array.isRequired
};

export default CardArea;