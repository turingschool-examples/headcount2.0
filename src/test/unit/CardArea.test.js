import React from 'react';
import { shallow } from 'enzyme';
import CardArea from '../../CardArea/CardArea';

describe('CardArea', () => {
  it('should match snapshot', () => {
    const cleanData = [{
      location: 'COLORADO',
      stats: [{2007: 1}]
    }];
    const cardArea = shallow(<CardArea cleanData={cleanData} selectedCards={[]} selectCard={jest.fn()}/>);

    expect(cardArea).toMatchSnapshot();
  });
});
