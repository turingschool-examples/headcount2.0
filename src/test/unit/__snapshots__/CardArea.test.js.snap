// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardArea should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardArea
    cleanData={
        Array [
            Object {
              "location": "COLORADO",
              "stats": Array [
                Object {
                  "2007": 1,
                },
              ],
            },
          ]
    }
    selectCard={[Function]}
    selectedCards={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          location="COLORADO"
          selectCard={[Function]}
          selectedCards={Array []}
          stats={
                    Array [
                              Object {
                                "2007": 1,
                              },
                            ]
          }
/>,
      ],
      "className": "card-area",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "location": "COLORADO",
          "selectCard": [Function],
          "selectedCards": Array [],
          "stats": Array [
            Object {
              "2007": 1,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            location="COLORADO"
            selectCard={[Function]}
            selectedCards={Array []}
            stats={
                        Array [
                                    Object {
                                      "2007": 1,
                                    },
                                  ]
            }
/>,
        ],
        "className": "card-area",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "location": "COLORADO",
            "selectCard": [Function],
            "selectedCards": Array [],
            "stats": Array [
              Object {
                "2007": 1,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
